### DISCUSSION TITLE
How to fill solid color in a shape?

### DISCUSSION THREAD

**Discussion by pavitrashah** [mozman/ezdxf#796] [Q&A] [answered]
Hello,

I am trying to convert dxf files into png images.
The files will only have a single shape in them and I want to be able to fill the shape with solid color.

> The code I've tried to achive this.

```
class MyFrontend(Frontend):
    def override_properties(self, entity: DXFGraphic, properties: Properties) -> None:
        properties.color = '#FF0000'
        properties.filling = 'SOLID'
```

The result I'm having with this code is like this:
![im](https://user-images.githubusercontent.com/47504253/207641864-2d1bb875-9eb2-4d17-9b25-be982a8e3f34.jpg)

**Comment by mozman** [comment, marked as answer]
Only certain DXF entities support filling: SOLID, TRACE and HATCH.

A tutorial how to create HATCH entities can be found [here](https://ezdxf.mozman.at/docs/tutorials/hatch.html).

**Reply by pavitrashah** [reply]
In the tutorial it is showing how to create a new shape with hatch.

But I'm getting the shapes already made in my case in dxf file and it can be any thing. A eclipse or even a christmas tree. So how can I convert them to have them filled with hatch property?

**Reply by mozman** [reply]
Ezdxf does not provide such functionality, you need a proper CAD application for that.

**Reply by pavitrashah** [reply]
Any suggestions on which application or library might help to achieve this? 
Or if using a different backend can help to fill solid color in the shape.

**Reply by mozman** [reply]
AutoCAD, BricsCAD, ZwCAD, GStar CAD
